builtin/logical/database/rotation.go

```diff
@@ -64,6 +64,29 @@ func (b *databaseBackend) populateQueue(ctx context.Context, s logical.Storage)
 			continue
 		}
 
+		// If an account's NextVaultRotation period is nil, it means that the
+		// role was created before we added the `NextVaultRotation` field. In this
+		// case, we need to calculate the next rotation time based on the
+		// LastVaultRotation and the RotationPeriod. However, if the role was
+		// created with skip_import_rotation set, we need to use the current time
+		// instead of LastVaultRotation because LastVaultRotation is 0
+		if role.StaticAccount.NextVaultRotation.IsZero() {
+			log.Debug("NextVaultRotation is zero", roleName)
+			// Previously skipped import rotation roles had a LastVaultRotation value of zero
+			if role.StaticAccount.LastVaultRotation.IsZero() {
```

Comment:
So `IsZero()` can either man that the configuration has not been automatically updated or is literally `0` or both?

Reply:
Here, we are using IsZero() to check if `LastVaultRotation` has been set. I guess it also checks if it is 0, but from the code it seems like we only ever set LastVaultRotation to now(), so that shouldn't be a possibility. If `LastVaultRotation` hasn't been set it'll be the zero value of time.Time. We assume that if it's the zero value, then it's because it was created with `skip_import_rotation=true` which means we don't rotate the static-role on creating, so we don't know when the LastVaultRotation was. 

However, I think the scheduling logic (`rotation_schedule`), might also touch these fields so i'll double check.

Reply:
I audited rotation_schedule logic, and it works similarly so I think we are safe here!